extends Node2D

@export var enemy_bee_scene: PackedScene
@export var max_enemy := 10

var current_enemy := 0
@export var spawn_delay := 3.0

# Called every frame. 'delta' is the elapsed time since the previous frame.
# Physics/Movement/Input â†’ _physics_process()
# Visuals/UI/Animation â†’ _process()

#_physics_process(delta)
#Cháº¡y vá»›i tá»‘c Ä‘á»™ cá»‘ Ä‘á»‹nh (máº·c Ä‘á»‹nh 60 FPS)
#ÄÆ°á»£c Ä‘á»“ng bá»™ vá»›i physics engine
#delta luÃ´n á»•n Ä‘á»‹nh vÃ  cÃ³ thá»ƒ dá»± Ä‘oÃ¡n Ä‘Æ°á»£c
#ThÃ­ch há»£p cho: váº­t lÃ½, di chuyá»ƒn nhÃ¢n váº­t, va cháº¡m, input xá»­ lÃ½
#
#_process(delta)
#Cháº¡y má»—i frame (tá»‘c Ä‘á»™ thay Ä‘á»•i tÃ¹y FPS)
#delta thay Ä‘á»•i theo hiá»‡u nÄƒng mÃ¡y
#KhÃ´ng Ä‘á»“ng bá»™ vá»›i physics
#ThÃ­ch há»£p cho: animation, UI, hiá»‡u á»©ng visual, camera
func _process(_delta: float) -> void:
	pass


func _on_fire_button_pressed() -> void:
	$player.shoot()
	pass # Replace with function body.


func _on_fire_button_2_pressed() -> void:
	$player.shoot_base_bullet()

func _ready():
	randomize()
	spawn_loop()

func spawn_loop():
	while true:
		await get_tree().create_timer(spawn_delay).timeout
		spawn_enemy()

func spawn_enemy():
	if current_enemy >= max_enemy:
		return
		
	var enemy = enemy_bee_scene.instantiate()
	# ğŸ”¥ RANDOM Vá»Š TRÃ TRONG MÃ€N HÃŒNH
	var screen_size = get_viewport_rect().size
	var x = randf_range(0, screen_size.x)
	var y = randf_range(0, screen_size.y * 0.6) # trÃ¡nh UI dÆ°á»›i

	enemy.global_position = Vector2(x, y)

	enemy.tree_exited.connect(func(): current_enemy -= 1)
	current_enemy += 1
	add_child(enemy)
